internal/transactionsubmission/engine/signing/distribution_account_resolver.go

```diff
@@ -60,32 +62,36 @@ type DistributionAccountResolverImpl struct {
 }
 
 // DistributionAccount returns the tenant's distribution account stored in the database.
-func (r *DistributionAccountResolverImpl) DistributionAccount(ctx context.Context, tenantID string) (string, error) {
-	tnt, err := r.tenantManager.GetTenantByID(ctx, tenantID)
-	if err != nil {
-		return "", fmt.Errorf("getting tenant by ID: %w", err)
-	}
-
-	if tnt.DistributionAccount == nil {
-		return "", ErrDistributionAccountIsEmpty
-	}
-
-	return *tnt.DistributionAccount, nil
+func (r *DistributionAccountResolverImpl) DistributionAccount(ctx context.Context, tenantID string) (*schema.DistributionAccount, error) {
+	return r.getDistributionAccount(func() (*tenant.Tenant, error) {
```

Comment:
❓I'm sure there is a good reason for this but I don't see it. Why are we passing a method that resolves the tenant instead of resolving the tenant and passing it here ?

Reply:
Good catch, that's not needed. Simplified in https://github.com/stellar/stellar-disbursement-platform-backend/pull/280/commits/fb48fb0b979ddb75d6c7844ee7afa5409999b99e